pkg/controller/signup_test.go

```diff
@@ -73,75 +66,105 @@ func (s *TestSignupSuite) TestSignupGetHandler() {
-
 	// Create a mock SignupService
-	svc := &signup.ServiceImpl{
-		Namespace:   "test-namespace-123",
-		UserSignups: fake.NewFakeUserSignupClient("test-namespace-123"),
-		MasterUserRecords: fake.NewFakeMasterUserRecordClient("test-namespace-123"),
-	}
+	svc := &FakeSignupService{}
 	// Create UserSignup
-	userID, err := uuid.NewV4()
-	require.NoError(s.T(), err)
-	userSignup, err := svc.CreateUserSignup("jsmith", userID.String())
-	require.NoError(s.T(), err)
-	require.NotNil(s.T(), userSignup)
+	userID := uuid.NewV4().String()
 
-	// Create SignupCheck check instance.
-	SignupCheckCtrl := controller.NewSignup(logger, s.Config, svc)
-	handler := gin.HandlerFunc(SignupCheckCtrl.GetHandler)
+	// Create Signup controller instance.
+	ctrl := controller.NewSignup(logger, s.Config, svc)
+	handler := gin.HandlerFunc(ctrl.GetHandler)
 
-	s.Run("SignupCheck for not ready signups", func() {
+	s.Run("signups found", func() {
 		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
 		rr := httptest.NewRecorder()
 		ctx, _ := gin.CreateTestContext(rr)
 		ctx.Request = req
+		ctx.Set(middleware.SubKey, userID)
+
+		expected := &signup.Signup{
+			TargetCluster: uuid.NewV4().String(),
+			Username:      "jsmith",
+			Status: signup.Status{
+				Reason: "Provisioning",
+			},
+		}
+		svc.MockGetUserSignup = func(id string) (*signup.Signup, error) {
+			if id == userID {
+				return expected, nil
+			}
+			return nil, nil
+		}
 
-		ctx.Set(middleware.SubKey, userID.String())
-		ctx.Set(middleware.EmailKey, "email@email.email")
-		ctx.Set(middleware.UsernameKey, "jsmith")
 		handler(ctx)
 
 		// Check the status code is what we expect.
-		assert.Equal(s.T(), rr.Code, http.StatusOK, "handler returned wrong status code: got %v want %v", rr.Code, http.StatusOK)
-
-		log.Println(string(rr.Body.Bytes()))
+		assert.Equal(s.T(), http.StatusOK, rr.Code, "handler returned wrong status code")
 
 		// Check the response body is what we expect.
 		data := &signup.Signup{}
 		err := json.Unmarshal(rr.Body.Bytes(), &data)
 		require.NoError(s.T(), err)
 
-		val := data.Status.Ready
-		assert.Equal(s.T(), apiv1.ConditionFalse, val, "ProvisioningDone is true in test mode signupcheck initial response")
+		assert.Equal(s.T(), expected, data)
 	})
 
-	s.Run("SignupCheck for ready signups", func() {
-		// Set mock signup to ready
-		// TODO: this needs to be implemented
+	s.Run("signups not found", func() {
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		rr := httptest.NewRecorder()
+		ctx, _ := gin.CreateTestContext(rr)
+		ctx.Request = req
+		ctx.Set(middleware.SubKey, userID)
+
+		svc.MockGetUserSignup = func(id string) (*signup.Signup, error) {
+			return nil, nil
+		}
 
+		handler(ctx)
+
+		// Check the status code is what we expect.
+		assert.Equal(s.T(), http.StatusNotFound, rr.Code, "handler returned wrong status code")
```

Comment:
For some reason our controller doesn't set `http.StatusNotFound` properly. So this test is currently failing. But it seems to be a bug in our code and the test just caught it.

Reply:
It was a bug in the controller. Fixed in https://github.com/michaelkleinhenz/registration-service/pull/2/commits/2f846143eaf181feb88006ef691bf7f37cf3e568